src/name.rs

```diff
@@ -399,6 +399,24 @@ pub(crate) struct NamespaceResolver {
     nesting_level: i32,
 }
 
+/// These constants define the reserved namespaces for the xml standard.
+///
+/// The prefix `xml` is by definition bound to the namespace name
+/// `http://www.w3.org/XML/1998/namespace`. It may, but need not, be declared, and must not be
+/// undeclared or bound to any other namespace name. Other prefixes must not be bound to this
+/// namespace name, and it must not be declared as the default namespace.
+///
+/// The prefix `xmlns` is used only to declare namespace bindings and is by definition bound
+/// to the namespace name http://www.w3.org/2000/xmlns/. It must not be declared or
+/// undeclared. Other prefixes must not be bound to this namespace name, and it must not be
+///  declared as the default namespace. Element names must not have the prefix xmlns.
+///
+/// [reserved namespaces]: https://www.w3.org/TR/xml-names11/#xmlReserved
+static WELL_KNOWN_NAMESPACES: &'static [[&str; 2]] = &[
```

Comment:
I'd just call it `RESERVED_NAMESPACES`

Reply:
I'm planning on moving the logic depending on our agreement on the error behavior. I'm thinking this change may not use this constant, so I am going to try to get agreement on the error behavior and then rewrite with that in mind.